src/flakeModule.nix

```
@@ -0,0 +1,167 @@
+{
+  validators,
+  grow,
+  pick,
+  winnow,
+  harvest,
+}: {
+  inputs,
+  config,
+  options,
+  lib,
+  ...
+}: let
+  inherit
+    (builtins)
+    all
+    concatMap
+    head
+    isList
+    length
+    mapAttrs
+    tail
+    zipAttrsWith
+    ;
+  inherit
+    (lib)
+    getValues
+    mkOption
+    mkOptionType
+    showDefs
+    showOption
+    types
+    literalExpression
+    ;
+
+  # ensures we always end up with a list of lists
+  # if more than one definition
+  mergeHarvesterOption = loc: defs: let
+    list = getValues defs;
+  in
+    if length list == 1
+    then head list
+    else if all isList list
+    then
+      concatMap (l:
+        if isList l
+        then l
+        else [l])
+      list
+    else throw "Cannot merge definitions of `${showOption loc}'. Definition values:${showDefs defs}";
+
+  harvesterType = mkOptionType {
```

```suggestion
  harvesterType = mkOptionType {
    name = "harvester";
```

This name is internal to the module system, but required for the documentation to eval. Odd.